utils/retry/retry.go

```diff
@@ -0,0 +1,24 @@
+package retry
+
+import (
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+// BackoffExponential makes maxRetry attempts to executed function f. After the nth attempt,
+// we wait retryMilliseconds*2^n ms before retrying. Returns an error after
+// maxRetry unsuccessful attempts.
+func BackoffExponential(f func() error, maxRetry int, retryMilliseconds int, logger zerolog.Logger) bool {
```

Comment:
We would need ability to cancel this retries, ie, in case of node shutdown.
That would probably require this to be turned into a component with `ReadyDoneAware` quantities.

Reply:
We can add that in a separate ticket and PR , this retry for now solves our issue for retrying messages in a tight loop.

Reply:
I am in the process if adding a func RetryWithContext . This is more code, but cleaner.